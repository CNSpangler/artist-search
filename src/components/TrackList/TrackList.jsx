import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const TrackList = ({ artistName, tracks }) => {
  const trackElements = tracks.map(track => (
    <li key={track.songId}>
      <Link to={{
        pathname: `/tracks/${artistName}/${track.songId}`,
        songTitle: track.songTitle
      }} >
        <span>{track.songTitle}</span>
      </Link>
    </li>
  ));

  return (
    <ol>
      {trackElements}
    </ol>
  );
};

TrackList.propTypes = {
  artistName: PropTypes.string.isRequired,
  tracks: PropTypes.arrayOf(PropTypes.shape({
    songId: PropTypes.string.isRequired,
    songTitle: PropTypes.string.isRequired
  }))
};

export default TrackList;
