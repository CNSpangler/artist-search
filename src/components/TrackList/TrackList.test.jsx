import React from 'react';
import { shallow } from 'enzyme';
import TrackList from './TrackList.jsx';

describe('TrackList component', () => {
  it('renders TrackList', () => {
    const wrapper = shallow(<TrackList releases={[
      {
        songId: 1234,
        songTitle: 'Fave Song'
      },
      {
        songId: 1235,
        songTitle: 'Hips Don\'t Lie'
      }
    ]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
