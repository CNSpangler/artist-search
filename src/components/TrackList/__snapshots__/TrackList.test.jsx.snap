// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList component renders TrackList 1`] = `<ol />`;
